### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by carolinapowers**

### CODE REVIEW

**Comment by carolinapowers** [comment]
@al-the-x  Here is our new repo... We lost some code when refactoring our file tree, which made us migrate to this repo. 

So, we were trying to Login with the Petowner button and being redirected to the Owner's page. 

In order to redirect the Petsitter to the sitter's page, we used the $state.go('client') in the Factory and it works well. 

Now, we are trying to use the $state.go or even the $transitionTo inside the 'OwnerBtnCtrl'  to redirect the Owner to Owner's page. But it won't work... We can get it work if we write the login function inside the Controller and use the $state.go('owner') . However, based on what I have heard, the login function should be inside a Factory. 

If the login function does need to be inside a Factory, how could we redirect the Owner to his page after importing the Auth.login inside the 'OwnerBtnCtrl'? 

Any help would be appreciated.

**Comment by carolinapowers** [comment]
Hmmmm... But I'm thinking that if owner only has one page, and he will only read that page. Perhaps, it shouldn't be a problem to have owner logging in with a Controller? Does it make sense?

File: src/app/scripts/index.js [hunk 1]
```diff
@@ -18,7 +18,123 @@ angular.module('whatsPup', ['ngAnimate', 'ngCookies', 'ngTouch', 'ngSanitize', '
...
+    .state('owner', {
+    url: '/owner',
+    templateUrl: 'app/view/ownerpage.html',
+    controller: 'OwnerBtnCtrl',
+    controllerAs: 'ownerbtn'
+})
     $urlRouterProvider.otherwise('/');
+})
+
+.factory('Auth', function ($firebaseObject, $state) {
```
Comment (al-the-x) [inline src/app/scripts/index.js:39]: Would be a lot easier to read if this `Auth` service was defined in a separate file...

File: src/app/scripts/index.js [hunk 2]
```diff
@@ -18,7 +18,123 @@ angular.module('whatsPup', ['ngAnimate', 'ngCookies', 'ngTouch', 'ngSanitize', '
...
+    return {
+        /**
+         * Wrapper for `onAuth` that filters the `auth` object
+         * through the `updateUser()` function
+         */
+        onAuth: function (creds) {
+            auth.onAuth(function (data) {
+                creds(updateUser(data));
+            });
+        },
```
Comment (al-the-x) [inline src/app/scripts/index.js:52]: Can you point me to where you are using this code...?

File: src/app/scripts/controllers/ownerbutton.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+        Auth.onAuth(function (user) {
+            self.user = user;
+            if (user === null) {
+                console.log('null')
+            } else {
+                console.log(user)
+            }
+        });
```
Comment (al-the-x) [inline src/app/scripts/controllers/ownerbutton.js:72]: This is the only use of `Auth.onAuth`...?

**Comment by jaredstevick** [comment]
@al-the-x This is the only use for it at the moment, as per Logan's suggestion.  We're still a little fuzzy on the whole onAuth thing though. We have read through the firebase documentation but we haven't had much luck.  Do you have any suggestions for other resources that may help us?

**Comment by al-the-x** [comment]
I think you're getting tripped up in the complexity of that `Auth` service. I can tell exactly where that code originated. You might have an easier time seeing your way out by re-implementing that as a single big procedure... Diagram it. Figure out what it's doing. Then figure out what you need.

**Comment by al-the-x** [comment]
Remember that Authentication and Authorization are _separate_ concerns. Figuring out who someone is -- their Facebook identity, for example -- comes first. _Then_ identify them as a Sitter or an Owner, logically. You might collapse that into a single step in the User Interface, but your code should still take those individual steps in that order.

**Pull request merged**